tests/system/action/test_internal_actions.py

```diff
@@ -89,21 +83,53 @@ def test_internal_no_password_in_request(self) -> None:
         self.assert_status_code(response, 401)
         self.assert_model_not_exists("user/2")
 
-    @patch("openslides_backend.shared.env.is_dev_mode")
-    def test_internal_no_password_on_server(self, is_dev_mode: Mock) -> None:
+    @patch("openslides_backend.shared.env.Environment.is_dev_mode")
+    def test_internal_no_password_on_server(self, is_dev_mode: MagicMock) -> None:
         is_dev_mode.return_value = False
-        del os.environ[INTERNAL_AUTH_PASSWORD_FILE]
-        response = self.internal_request("user.create", {"username": "test"})
+        response = self.internal_request(
+            "user.create", {"username": "test"}, "some password"
+        )
         self.assert_status_code(response, 500)
         self.assert_model_not_exists("user/2")
 
-    @patch("openslides_backend.shared.env.is_dev_mode")
+
+class TestInternalActionsProd(BaseInternalActionsTest):
+    """
+    That same as the TestInternalActionsDev class but we patch the is_dev_mode
+    in every test.
+    """
+
+    def setUp(self) -> None:
+        self.internal_auth_password = "my secret password"
+        self.secret_file = NamedTemporaryFile()
+        self.secret_file.write(self.internal_auth_password.encode("ascii"))
+        self.secret_file.seek(0)
+        os.environ["INTERNAL_AUTH_PASSWORD_FILE"] = self.secret_file.name
+        super().setUp()
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        del os.environ["INTERNAL_AUTH_PASSWORD_FILE"]
+        self.secret_file.close()
+
+    @patch("openslides_backend.shared.env.Environment.is_dev_mode")
+    def test_internal_try_access_backend_internal_action_wrong_pw(
+        self, is_dev_mode: MagicMock
+    ) -> None:
+        is_dev_mode.return_value = False
+        response = self.internal_request(
+            "user.create", {"username": "my username"}, "wrong pw"
+        )
+        self.assert_status_code(response, 401)
+        self.assert_model_not_exists("user/2")
+
+    @patch("openslides_backend.shared.env.Environment.is_dev_mode")
     def test_internal_try_access_backend_internal_action(
```

Comment:
This test is wrong now: Backend-internal actions should only be accessed from inside the backend (and, for testing purposes, in dev mode from the outside). But in production mode, they can never be accessed from the outside which this test previously reflected. I don't know why this suddenly succeeds since the check in `action_hanlder.py` is still present, but please fix this before merging.

Reply:
Sorry, I do not know how to test that the route can not be accessed from outside. I even do not know how the backend should know what is "inside" and "outside". I think we have the proxy config for this. So I think the test is correct. It shows that one can reach the internal action using the correct password.

Reply:
Sorry, I somehow missed that you also changed the called action from `option.create` to `user.create`. So, yes, this test is correct, but we still also need the old test with `option.create`.

There are 3 types of backend actions (see also https://github.com/OpenSlides/openslides-backend/blob/2c1493c2c10143d9ed0d9a6a5fc362ff5169eced/openslides_backend/action/util/action_type.py): "normal" or "public" actions which are accessible from the "outside" (i.e. the client), "stack internal" actions which are only accessible from inside the stack (i.e. the manage service) and "backend internal" which should not be accessed from anyone else but the backend and are only used for internal operations. This test tests that a "backend internal" action cannot be accessed from the outside, even from inside the stack with the correct password. "Outside meaning here via HTTP.

So, please add back the previous test, which asserted that `option.create` cannot be called from outside of the backend.

Reply:
Done.